src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs

```diff
@@ -21,11 +22,24 @@ public class JobConversions
     {
         #region Hydra to PS convertors
 
+        /// <summary>
+        /// This function returns either job object or job details object based on 
+        /// what hydra object contains.
+        /// To elaborate, if hydra job's ExtendedInfo is filled then this function will
+        /// return a job details object. Otherwise it will return a job object.
+        /// </summary>
+        /// <param name="hydraJob"></param>
+        /// <returns></returns>
         public static AzureRmRecoveryServicesJobBase GetPSJob(JobResponse hydraJob)
         {
+            return GetPSJob(hydraJob.Item);
+        }
+
+        public static AzureRmRecoveryServicesJobBase GetPSJob(JobResource hydraJob)
+        {
             AzureRmRecoveryServicesJobBase response = null;
 
-            if (hydraJob.Item.Properties.GetType() == typeof(AzureIaaSVMJob))
+            if (hydraJob.Properties.GetType() == typeof(AzureIaaSVMJob))
             {
                 response = GetPSAzureVmJob(hydraJob);
             }
```

Comment:
Good catch. Actually Properties field will be null when a job of another type is sent by service. This will throw exception. I will handle that case and trace a warning/info.